src/SwedbankPay.Sdk/PaymentOrders/PaymentOrder.cs

```diff
@@ -0,0 +1,122 @@
+                        break;
+                    case PaymentOrderResourceOperations.UpdatePaymentOrderUpdateOrder: 
+                        operations.Update = new ExecuteRequestWrapper<PaymentOrderUpdateRequestContainer, PaymentOrderResponseContainer>(
+                            httpOperation.Request, client, m => new CouldNotUpdatePaymentOrderException(httpOperation.Href, m));
+                        break;
+                    case PaymentOrderResourceOperations.UpdatePaymentOrderAbort:
+                        operations.Abort = new ExecuteWrapper<PaymentOrderResponseContainer>(
+                            httpOperation.Request, client, m => new CouldNotPostTransactionException(httpOperation.Href, m),
+                            new PaymentAbortRequestContainer());
+                        break;
+                    case "view-paymentorder":
+                        operations.View = httpOperation;
+                        break;
+                }
+
+                Operations = operations;
+            }
+           
+        }
+
+
+        public Operations Operations { get; }
+        public PaymentOrderResponse PaymentOrderResponse { get; }
+        
+        internal static async Task<PaymentOrder> Create(PaymentOrderRequest paymentOrderRequest,
+                                                        SwedbankPayHttpClient client, PaymentOrderExpand paymentOrderExpand = PaymentOrderExpand.None)
+        {
+            var url = $"/psp/paymentorders{GetExpandQueryString(paymentOrderExpand)}";
+            //paymentOrderRequest.SetRequiredMerchantInfo(swedbankPayOptions);
+
+            var payload = new PaymentOrderRequestContainer(paymentOrderRequest);
+
+            Exception OnError(ProblemsContainer m)
+            {
+                return new CouldNotPlacePaymentOrderException(payload, m);
+            }
+
+            var paymentOrderResponseContainer =
+                await client.HttpRequest<PaymentOrderResponseContainer>(HttpMethod.Post, url, OnError, payload);
+
+            return new PaymentOrder(paymentOrderResponseContainer, client);
+        }
+
+
+        internal static async Task<PaymentOrder> Get(string id,
+                                                     SwedbankPayHttpClient client, PaymentOrderExpand paymentOrderExpand = PaymentOrderExpand.None)
+        {
+            if (string.IsNullOrEmpty(id))
+                throw new ArgumentNullException(nameof(id), $"{id} cannot be null or empty");
+
+            var url = $"{id}{GetExpandQueryString(paymentOrderExpand)}";
+
+            Exception OnError(ProblemsContainer m)
+            {
+                return new CouldNotFindPaymentException(id, m);
+            }
+
+            var paymentOrderResponseContainer = await client.HttpGet<PaymentOrderResponseContainer>(url, OnError);
+
+            return new PaymentOrder(paymentOrderResponseContainer, client);
+        }
+
+
+        internal static string GetExpandQueryString<T>(T expandParameter)
```

Comment:
PaymentOrderExpand is the only enum being used here so consider using that as the input type instead of generic T.

Reply:
Or even better, don't use an enum with `[Flags]` and instead go for the solution suggested by @asbjornu above.

Reply:
Changed to use the GetExpandQueryString from ResourceBase instead of the method in PaymentOrder. So it's still generic but uses more than PaymentOrderExpand.